firmware/application/apps/ui_playlist.cpp

```diff
@@ -198,17 +213,41 @@ void PlaylistView::stop(const bool do_loop) {
     if (is_active()) {
         replay_thread.reset();
     }
-    if (do_loop) {
-        if (playlist_db.size() > 0) {
-            start();
+
+    // TODO: the logic here could be more beautiful but maybe they are all same for compiler anyway....
+    // Notes of the logic here in case if it needed to be changed in the future:
+    // 1. check_loop.value() is for the LOOP checkbox
+    // 2. do_loop is a part of the replay thread, not a user - control thing.
+    // 3. when (total_tracks != track_number) is true, it means that the current track is not the last track.
```

Comment:
Usually better to use >= instead of != when checking length. If you mess up, you can end up stuck in the loop in the != case.

Reply:
Thank you! fixed.